### PR TITLE
CI: automatically update the repo every 5 days

### PR DESCRIPTION
**Pull request by Williangalvani**
This seems to be deleting some of Copter 3.5 files? maybe it is the python version? maybe I'm missing some dependency?
see https://github.com/Williangalvani/ParameterRepository/commit/b3d6b52482f90b4287b1a7a07ffd963d91d83650

### CODE REVIEW

**Comment by clydemcqueen** [comment]
Looks like a bug in the json emitter in `param_parse.py`.

~~~
clyde@topside:~/ap/upstream (Copter-4.1)$ ./Tools/autotest/param_metadata/param_parse.py --vehicle ArduCopter --format all
clyde@topside:~/ap/upstream (Copter-4.1)$ ll -t
total 5664
-rw-rw-r--   1 clyde clyde  961282 Feb 26 11:42 ParameterMetaData.xml
-rw-rw-r--   1 clyde clyde  595035 Feb 26 11:42 Parameters.md
-rw-rw-r--   1 clyde clyde 1748182 Feb 26 11:42 Parameters.rst
-rw-rw-r--   1 clyde clyde  991226 Feb 26 11:42 Parameters.html
-rw-rw-r--   1 clyde clyde 1198335 Feb 26 11:42 apm.pdef.xml
-rw-rw-r--   1 clyde clyde    1948 Feb 26 11:42 apm.pdef.json
drwxrwxr-x  18 clyde clyde    4096 Feb 26 11:36 ./
drwxrwxr-x   8 clyde clyde    4096 Feb 26 11:35 .git/
drwxrwxr-x   2 clyde clyde    4096 Feb 26 11:35 mk/
drwxrwxr-x   9 clyde clyde    4096 Feb 26 11:35 modules/
...
~~~

I see `json_from_xml.py` -- perhaps this is a known bug?

**Comment by clydemcqueen** [comment]
Addresses #2

**Comment by Williangalvani** [comment]
@patrickelectric thoughts?

**Comment by HTRamsey** [comment]
What's the plan here? I'd like to update QGC with either this or just use the run_param_parse.py tool separately.

**Comment by patrickelectric** [comment]
I'll investigate what is going on

**Comment by patrickelectric** [comment]
This should be fixed by https://github.com/ArduPilot/ParameterRepository/pull/5

File: scripts/run_param_parse.py [hunk 1]
```diff
@@ -92,14 +92,16 @@ def run(self):
             except Exception as exception:
                 print(exception)
 
-            output = f'/tmp/output'
+            output = '.'
```
Comment (patrickelectric) [inline scripts/run_param_parse.py]: To be safe, this should use:
```suggestion
            output = Path(__file__).parent.parent
```

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          python-version: '3.8' # Set the Python version you need
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip GitPython packaging lxml emit
+          # Add any other dependencies here
+
+      - name: Run Python script
+        run: python scripts/run_param_parse.py
```
Comment (patrickelectric) [inline .github/workflows/deploy.yaml]: jsom_from_xml also needs to run, the problem is that old firmwares failed on generate json files correctly.

**Review by patrickelectric — CHANGES_REQUESTED** [review]

**Comment by patrickelectric** [comment]
Please update over master, should contain all necessary fixes for this patch to work

**Comment by HTRamsey** [comment]
@patrickelectric thanks for looking into this so quickly

File: .github/workflows/deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+      - name: Commit and push if changed
+        run: |
+          git config --global user.email "CI@example.com"
+          git config --global user.name "CI"
```
Comment (patrickelectric) [inline .github/workflows/deploy.yaml]: Please use the official git bot:
```auggestion
        git config --global user.name 'Git bot'
        git config --global user.email 'bot@noreply.github.com'
```

File: .github/workflows/deploy.yaml (hunk 2 above)
Comment (patrickelectric) [reply .github/workflows/deploy.yaml]: Where is json_from_xml being called ?

Comment (Williangalvani) [reply .github/workflows/deploy.yaml]: I assumed we didnt need that anymore after the previous prs. I'll check this tomorrow

**Comment by patrickelectric** [comment]
@Williangalvani it appears that there is something wrong on the PR , is deleting copter, plane and sub 4.0. It's also weird to be modifying the files, since it was updates a couplem of hours ago from the last version

**Pull request merged**